benchmarks/foveation/main.cpp

```diff
@@ -0,0 +1,771 @@
+    mKnobs.pRenderPattern->SetValidator([](const std::string& res) {
+        return res == "" ||
+               res == "clear" ||
+               res == "constant" ||
+               res == "static-noise" ||
+               res == "dynamic-noise";
+    });
+
+    GetKnobManager().InitKnob(&mKnobs.pFoveationPattern, "foveation-pattern", "");
+    mKnobs.pFoveationPattern->SetFlagDescription("Set the foveation pattern used for rendering ('1x1', '2x2', '4x4', 'radial', 'anisotropic').");
+    mKnobs.pFoveationPattern->SetValidator([](const std::string& res) {
+        return res == "" ||
+               res == "1x1" ||
+               res == "2x2" ||
+               res == "4x4" ||
+               res == "radial" ||
+               res == "anisotropic";
+    });
+
+    GetKnobManager().InitKnob(&mKnobs.pSubsampledImage, "enable-subsampled-image", kDefaultEnableSubsampledImage);
+    mKnobs.pSubsampledImage->SetFlagDescription("Enable the subsampled image flag on the render target.");
+
+    GetKnobManager().InitKnob(&mKnobs.pRenderResolution, "render-resolution", std::make_pair(kDefaultWidth, kDefaultHeight));
+    mKnobs.pRenderResolution->SetFlagDescription("Width and height of render target in pixels.");
+    mKnobs.pRenderResolution->SetFlagParameters("<width>x<height>");
+
+    GetKnobManager().InitKnob(&mKnobs.pPostResolution, "post-resolution", std::make_pair(kDefaultWidth, kDefaultHeight));
+    mKnobs.pPostResolution->SetFlagDescription("Width and height of post render target in pixels.");
+    mKnobs.pPostResolution->SetFlagParameters("<width>x<height>");
+
+    GetKnobManager().InitKnob(&mKnobs.pShowFragmentSize, "show-fragment-size", kDefaultShowFragmentSize);
+    mKnobs.pShowFragmentSize->SetFlagDescription("Show the fragment width and height in the G and B color channels.");
+
+    GetKnobManager().InitKnob(&mKnobs.pRenderScreenshotPath, "render-screenshot-path", "");
+    mKnobs.pRenderScreenshotPath->SetFlagDescription("Set the path to save a copy of the render image when a screenshot is triggered. By default the render target image will not be saved.");
+
+    GetKnobManager().InitKnob(&mKnobs.pPostScreenshotPath, "post-screenshot-path", "");
+    mKnobs.pPostScreenshotPath->SetFlagDescription("Set the path to save a copy of the post image when a screenshot is triggered. By default the render target image will not be saved.");
+
+    GetKnobManager().InitKnob(&mKnobs.pExtraHashRounds, "extra-hash-rounds", 0);
+    mKnobs.pExtraHashRounds->SetFlagDescription("Number of extra hash rounds to execute in the fragment shader.");
+}
+
+void FoveationBenchmarkApp::Config(ppx::ApplicationSettings& settings)
+{
+    settings.appName                            = "foveation_benchmark";
+    settings.enableImGui                        = false;
+    settings.grfx.api                           = kApi;
+    settings.grfx.swapchain.depthFormat         = grfx::FORMAT_D32_FLOAT;
+    settings.grfx.enableDebug                   = false;
```

Comment:
Can we remove this? It was in applications before, but I think it is better to let the bigwheels handle it, so it is on for debug mode for example

Reply:
Some of them are still needed: 
* If I remove kApi no api is assigned by default and hence we assert later
* If we do not ask for an specific depth format none is assigned.
I can remove ImGui and enable debug (since they default to false anyway) but I do not see value on it.
Is there an example of an app that does not set these?

Reply:
This comment was specifically for line 255. I think github shows the few lines before that. If you remove enableDebug it would use default behaviour. So if someone builds debug build on desktop it will use validation layers, which is preferrable to remember to run them

Reply:
Noted (that extra lines were shown due github interface)
Thanks